osu.Framework/Extensions/PopoverExtensions.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Cursor;
+
+#nullable enable
+
+namespace osu.Framework.Extensions
+{
+    public static class PopoverExtensions
+    {
+        /// <summary>
+        /// Shows the popover for <paramref name="hasPopover"/> on its nearest <see cref="PopoverContainer"/> ancestor.
+        /// </summary>
+        public static void ShowPopover(this IHasPopover hasPopover) => setTargetOnNearestPopover((Drawable)hasPopover, hasPopover);
+
+        /// <summary>
+        /// Hides the popover shown on <paramref name="drawable"/>'s nearest <see cref="PopoverContainer"/> ancestor.
+        /// </summary>
+        public static void HidePopover(this Drawable drawable) => setTargetOnNearestPopover(drawable, null);
+
+        private static void setTargetOnNearestPopover(Drawable origin, IHasPopover? target)
+        {
+            var node = origin.Parent;
+
+            while (node != null)
+            {
+                if (node is PopoverContainer popoverContainer)
+                {
+                    popoverContainer.SetTarget(target);
+                    return;
+                }
+
+                node = node.Parent;
+            }
+        }
```

Comment:
Nit: can benefit from the internal `FindClosestParent` method.

```suggestion
        private static void setTargetOnNearestPopover(Drawable origin, IHasPopover? target) => origin.FindClosestParent<PopoverContainer>()?.SetTarget(target);
```

Reply:
fair enough